### PR TITLE
vim-patch:8.2.3439,9.1.{0015,0021}

### PR DESCRIPTION
**Pull request by zeertzjq**
#### vim-patch:8.2.3439: deleted lines go to wrong yank register

Problem:    Deleted lines go to wrong yank register.
Solution:   Reset y_append when not calling get_yank_register(). (Christian
            Brabandt, closes vim/vim#8872)

https://github.com/vim/vim/commit/78eb9cce911dcd662182443e0920649ec1596d41

Nvim has removed y_append static variable and already works as expected.

Co-authored-by: Christian Brabandt <cb@256bit.org>

#### vim-patch:9.1.0015: i_CTRL-R- no longer works in replace mode

Problem:  i_CTRL-R- no longer works in replace mode
Solution: delete characters in replace mode before putting, add a test,
          add a bit warning into the documentation, that i_CTRL-R-P/O
          is not supported in Replace mode for now

closes: vim/vim#13816

https://github.com/vim/vim/commit/5d5cbb2b9ac526fb6fad2116e24a282affc45efe

Co-authored-by: Christian Brabandt <cb@256bit.org>

#### vim-patch:9.1.0021: i_CTRL-R- doesn't work for multibyte chars in Replace mode

Problem:  i_CTRL-R- doesn't work for multibyte chars in Replace mode,
          Coverity complains missing return value for u_save_cursor()
Solution: Use mb_charlen() and del_chars() instead, handle failure mode
          for u_save_cursor() correctly (@zeertzjq)

closes: vim/vim#13846

https://github.com/vim/vim/commit/424ec1f235a53131042aed94d30f6e2528e551d9

### CODE REVIEW

File: test/old/testdir/test_registers.vim [hunk 1]
```diff
@@ -838,5 +838,76 @@ func Test_replay_charsearch_omap()
   bwipe!
 endfunc
 
+" Make sure that y_append is correctly reset
```
Comment (tristan957) [inline test/old/testdir/test_registers.vim:841]: Drive-by review: Should the comment change given that this variable apparently no longer exists in Neovim according to the PR description?

Comment (zeertzjq) [reply test/old/testdir/test_registers.vim:841]: No
